### ISSUE TITLE
realm.delete([filtered array of Realm.ObjectClass]) no longer works

### ISSUE THREAD

**Issue by liamjones** [realm/realm-js#4327]
### Description

I'm testing the pre-release branch with our existing app. Removing a filtered list of `Realm.ObjectClass`es now throws an error but worked in 10.20.0 and earlier.

```typescript
    function removeFilteredThings() {
        const thingsToBeRemoved = realm
            .objects<RealmThing>(RealmThing.schema.name)
            .filter(realmThing => realmThing.toDomainObject().when.isBefore(LocalDateTime.now(clock)))
        realm.write(() => {
            realm.delete(thingsToBeRemoved)
        })
    }
```

The objects appear to be Realm objects according to the debugger (they have `__Realm_internal` and `_realm` properties. All of our properties show as `propName: Error: Exception in HostFunction: Realm accessed from incorrect thread.` but I'm assuming that's down to Safari debugger trying to access the prop while paused on the thrown exception inside the `realm.write()` callback?

### Stacktrace & log output

```shell
Error is: `Error: Exception in HostFunction: Argument to 'delete' must be a Realm object or a collection of Realm objects.`

JS stack trace should be irrelevant since it's all within our code.
```

### Can you reproduce a bug?

Yes, always

### Reproduction Steps

Let me know if you can't reproduce and need me to try and come up with a minimal repro.

### Version

hermes (10.20.0-beta.1)

### What SDK flavour are you using?

Local Database only

### Are you using encryption?

No, not using encryption

### Platform OS and version(s)

RN 0.66.3 using JSC (not Hermes!) - iPhone 13 Simulator w/ iOS 15.2, Android AVD Emulator running Android 9.0

**Comment by liamjones** [comment]
Hi @tomduncalf, thanks for the quick replies on this and #4328. I was afraid it was going to be more specific :/ I'll see what I can do about building a minimal repro or getting permission to share the app codebase with you in private.

**Comment by tomduncalf** [comment]
Yeah unfortunately it seems so! If it's any help, my usual approach in these "weird bug" cases is to strip my app down to the bare minimum – comment out all your code except e.g. this one Realm bit, so you have just a blank screen which is writing to Realm, remove as many dependencies as you can, then see if it works as expected. 

If it does work, then I gradually uncomment/add back in parts of the app until I find what is breaking it. If it does not work, then you can delete any commented out code and the app is now hopefully hopefully so minimal that you can share it with us without having much of your custom code in there at all!

**Comment by liamjones** [comment]
@fronck Oh, my mistake. Yes, I meant the non Hermes releases (<10.20.0)

**Comment by takameyer** [comment]
Hi @liamjones , if you want to share some code in private, feel free to send it to realm-help@mongodb.com.  We would definitely prefer a public example though, as this can help others who are having the same issue in the future.

**Comment by liamjones** [comment]
Hi @takameyer,

I've got permission to share a private repro and have emailed it in - see email with subject: "Private repro for realm-js issue #4327 (realm.delete([filtered array of Realm.ObjectClass]) no longer works in realm@hermes)".

Once the cause is understood (and assuming it isn't just a bug that gets fixed inside Realm) I'll try and write up a public explanation/example for others on the ticket.

**Comment by takameyer** [comment]
@liamjones It has arrived.  We will look at this as soon as we can.  Thanks for the assistance 🙂

**Comment by kneth** [comment]
@liamjones We have released v11.0.0, and we would like to know if you still observe the issue after upgrading.

**Comment by kneth** [comment]
@liamjones I am closing the issue. If you observe the behaviour after upgrading, please create a new issue.
